src/components/Subthoughts.tsx

```diff
@@ -405,17 +403,7 @@ export const SubthoughtsComponent = ({
   const editIndex = cursor && children && show ? children.findIndex(child => {
     return cursor[depth] && cursor[depth].rank === child.rank
   }) : 0
-  const filteredChildren = children.filter(child => {
-    const value = showContexts
-      ? head((child as ThoughtContext).context)
-      : (child as Child).value
-    return showHiddenThoughts ||
-      // exclude meta thoughts when showHiddenThoughts is off
-      // NOTE: child.rank is not used by isChildVisible
-      isChildVisible(state, pathToContext(simplePath), { value, rank: child.rank }) ||
-      // always include thoughts in cursor
-      (cursor && equalThoughtRanked(cursor[simplePath.length], child as Child))
-  })
+  const filteredChildren = filterChildren(state, resolvedPath, simplePath, children, showContexts)
```

Comment:
`children` has already filtered out visible thoughts, so there is some redundancy between the two. This was present before your addition, but it's worth pointing out. Ideally we could find a way to remove any redundancy between the original filtering where `children` is defined and the filtering here.